src/parse.rs

```diff
@@ -332,7 +490,31 @@ impl Chunk {
         let mut data = vec![0_u8; chunk_data_bytes];
         reader.read_exact(&mut data)?;
         *bytes_available -= chunk_size as i64;
-        Ok(Chunk { data, chunk_type })
+        Ok(Chunk { chunk_type, data })
+    }
+    fn read_all<R: Read>(
+        count: u32,
+        mut bytes_available: i64,
+        reader: &mut AseReader<R>,
+    ) -> Result<Vec<Self>> {
+        let mut chunks: Vec<Chunk> = Vec::new();
+        for _idx in 0..count {
+            let chunk = Self::read(&mut bytes_available, reader)?;
+            // Attach any user data chunks to the previously read chunk. Otherwise, push chunk to Vec and continue.
+            chunks.push(chunk);
+            // if let ChunkType::UserData = chunk.chunk_type {
+            //     let user_data = user_data::parse_userdata_chunk(&chunk.content.data)?;
```

Comment:
I'm guessing that's not needed anymore? And the comment above may also not be relevant anymore?

Reply:
Oh yes, good catch. Previously I did this a little differently (per the above comment) but moved to the chunk context field instead.